### PR TITLE
Arcgis services description

### PR DESCRIPTION
**Pull request by arx-it**
Add a description type to arcgis rest services browser list (ex: Feature Service or MapService)

### CODE REVIEW

**Comment by arx-it** [comment]
Yes
Picture1

**Comment by arx-it** [comment]
It add for each arcgis rest service if is a MapService or Feature Service

**Comment by nyalldawson** [comment]
Thanks for the contribution, but I'm a -1 to this change. I think it adds a lot of extra text to the browser just to show something which is already visible via the icon.

**Comment by arx-it** [comment]
Several entities have asked us for this change to quickly see the type of service, it is just the additional text in brackets (FeatureService or MapService)...

**Comment by nyalldawson** [comment]
What about putting this in the tooltip? 

The other issue I have is that some datasets are exposed as both feature services AND map services. QGIS shows two items for these under the same group. If you add the service type to the group, then it's misleading for these items...

**Comment by arx-it** [comment]
I will speak with my colleagues about it...
Keep in touch

**Comment by jlo-arxit** [comment]
Hi,
Quick follow-up on this PR after some discussion with the requestor.
We don't really understand your comments @nyalldawson (Aug 22), see our screenshot, it doesn't seem to us that both feature service and map service of a same dataset:
- have different icons > All services have the same kind of _database_ icon
- are grouped > All services are listed in a flat way under ArcGIS Server connexion

 ... or perhaps they now are in a latest QGIS version ? (sorry if it the case)

Anyway, if you think that adding some text in ( ) after service name make the UI too dense, we can indeed propose to use specific icons for feature / map / image services

**Review by arx-it — COMMENTED** [review]
Reopen pull request

**Comment by nyalldawson** [comment]
@arx-it 

Let's take an example:

https://gislegacy.scc.qld.gov.au/arcgis/rest/services/Biota/Biota_SCRC/MapServer


This is how it's shown in QGIS:

image

The datasets on the server have BOTH the "export map" and "query" capabilities, so they can be accessed using either the mapserver or feature server protocols. Accordingly it makes no sense to append either "Feature Server" or "Map Server" on the "Biota_SCRC" group, as the group contains both service types.

You can also see in the screenshot that different icons are used for mapserver (raster)/feature server (vector) versions of these layers.

**Comment by jlo-arxit** [comment]
@nyalldawson 
Thanks for your reply.
I think there is a confusion between ESRI "feature layers" (what you are talking about) and "feature/map services" (what we are talking about)

Your screenshot is showing the multiple **feature layers** inside a single **map service** called Biota_SCRC and reachable at the following endpoint :
https://gislegacy.scc.qld.gov.au/arcgis/rest/services/Biota/Biota_SCRC/MapServer
Note the end of the URL : "MapServer" means that you are targetting a single ArcGIS Server **service** 
image

What our development intend to adress is the distinction between m**ap services, feature services** and **image services** inside a global ArcGIS Server endpoint such as : https://xxx/arcgis/rest/services
This endpoint list all **services** available in the targetted ArcGIS Server, not only one. Please note this _database_ icon, same one for all **services** in the endpoint, whatever there type (mapserver / featureserver / imageserver)

Here is an example :

image

In QGIS : 
- for Biota map service, you are right, there is an icon distinction between the different feature **layers** of this single ArcGIS **map service** endpoint. The icon depends on how you want to use the feature layer : image/map or feature/query (in this second case, also an icon distinction based on geometry type)
- but our concern is one level higher, we need distinction between **map services**, **feature services** and **image services** when we are referencing a global ArcGIS rest endpoint in QGIS

image

Here an example of what we can do to differentiate map service, feature service and image services (quick work, surely need to rework icons) :
image

I hope it's clearer

**Comment by jlo-arxit** [comment]
A little _up_ before it gets closed ... 
Thanks

**Comment by jlo-arxit** [comment]
A little up before it gets closed ...
Thanks

**Comment by jlo-arxit** [comment]
Up

**Comment by jlo-arxit** [comment]
Up

**Comment by olivierdalang** [comment]
@jlo-arxit Hey ! You still have failures on the CI, I guess you'll have better chances getting another review and approval once fixed. From a quick look at the logs you probably miss some docstrings and haven't udated the SIP bindings (see https://docs.qgis.org/3.34/en/docs/developers_guide/codingstandards.html#generating-the-sip-file - I hope it's up to date, haven't used this in a long while)

**Comment by olivierdalang** [comment]
@jlo-arxit @arx-it Actually I read through the PR a bit too fast, I thought you had already implemented Nyall's suggestion. 

IMO Nyall's comment about using the tooltip rather than text in bracket is very valid and should be adressed, as indeed with the text it makes it the list harder to read.

About the icons, it's true that the difference is already visible at the layer level, but that requires unfolding the service to see the difference (plus for some reason, here we have duplicates under mapservices where layers both as vector and raster, which doesn't help, but that's a different issue).

For reference, here's how it looks in arcmap vs how it looks with the current implementation.

{AF2E7FFC-76D4-47E5-9101-2F58E0F9F84A}

image

To summarize, I'd suggest:
- revert the change to add the type in brackets
- insted show the type in a tooltip
- display different icons for the mapservices vs featureservices (both derived from the database icon, with a small subicon or whatever this is called, ideally inspired from what ArcMap shows).

@nyalldawson What do you think ? Would that work ?
